plugins/gitlab-catalog-backend/src/GitlabUserProcessor.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright SeatGeek
+ * Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+ */
+import { Entity, isUserEntity } from '@backstage/catalog-model';
+import { Config } from '@backstage/config';
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+import type {
+  CatalogProcessor,
+  CatalogProcessorEmit,
+} from '@backstage/plugin-catalog-node';
+import { ExpandedUserSchema, Gitlab } from '@gitbeaker/rest';
+import { Logger } from 'winston';
+
+const GITLAB_PER_PAGE_LIMIT = 500;
+const GITLAB_DEFAULT_HOST = 'https://gitlab.com';
+
+/**
+ * The GitlabUserProcessor is used to enrich our User entities with information
+ * from Gitlab: notably the user's Gitlab ID.
+ *
+ * @public
+ */
+export class GitlabUserProcessor implements CatalogProcessor {
```

Comment:
basically took what was defined in SlackUserProcessor and tweaked it to fit our needs for Gitlab